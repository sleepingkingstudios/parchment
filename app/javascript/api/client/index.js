import {
  applyMiddleware,
  selectMiddleware,
} from 'api/middleware/utils';
import {
  valueOrDefault,
} from 'utils/object';

import generateActions from './actions';
import generateHooks from './hooks';
import generateInitialState from './initialState';
import generateReducer from './reducer';
import generateRequest from './request';
import generateSelector from './selector';

const wrapReducer = ({ middleware, reducer }) => applyMiddleware(
  reducer,
  selectMiddleware(middleware, 'handleAction'),
);

const buildClient = (options) => {
  const {
    namespace,
    url,
  } = options;
  const data = valueOrDefault(options.data, {});
  const method = valueOrDefault(options.method, 'GET');
  const middleware = valueOrDefault(options.middleware, []);

  const actions = generateActions({ namespace });
  const initialState = generateInitialState({ data });
  const performRequest = generateRequest({
    actions,
    method,
    middleware,
    namespace,
    url,
  });
  const hooks = generateHooks({ performRequest });
  const reducer = wrapReducer({
    middleware,
    reducer: generateReducer({ actions, initialState }),
  });
  const selector = generateSelector({ namespace });

  return {
    actions,
    hooks,
    method,
    middleware,
    namespace,
    performRequest,
    reducer,
    selector,
    url,
  };
};

export default buildClient;
