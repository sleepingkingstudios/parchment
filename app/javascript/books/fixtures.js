import { load } from '../fixtures';

/* eslint-disable-next-line import/prefer-default-export */
export const booksData = load('books');
