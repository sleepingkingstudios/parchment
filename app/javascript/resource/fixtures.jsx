import React from 'react';

export const Block = () => (<div />);
export const Form = () => (<div />);
export const Table = () => (<div />);

export const CreatePage = () => (<div />);
export const IndexPage = () => (<div />);
export const LandingPage = () => (<div />);
export const ShowPage = () => (<div />);
export const UpdatePage = () => (<div />);

export const fixtures = {
  create: {
    component: CreatePage,
    exact: true,
    options: { baseUrl: '/widgets', resourceName: 'widgets' },
    path: '/create',
    reducer: ((state = { widget: {} }) => state),
  },
  index: {
    component: IndexPage,
    exact: true,
    options: { baseUrl: '/widgets', resourceName: 'widgets' },
    path: '/',
    reducer: ((state = { widgets: [] }) => state),
  },
  show: {
    component: ShowPage,
    exact: true,
    options: { baseUrl: '/widgets', resourceName: 'widgets' },
    path: '/:id',
    reducer: ((state = { widget: {} }) => state),
  },
  update: {
    component: UpdatePage,
    exact: true,
    options: { baseUrl: '/widgets', resourceName: 'widgets' },
    path: '/:id/update',
    reducer: ((state = { widget: {} }) => state),
  },
};
