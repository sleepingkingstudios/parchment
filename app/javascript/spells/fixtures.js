/* eslint-disable-next-line import/prefer-default-export */
export const spellsData = [
  {
    id: 'e17bca01-24d3-48d3-a308-099dd6dca373',
    level: 0,
    name: 'Embers',
    school: 'Evocation',
  },
  {
    id: 'e3641970-fc98-4674-b873-5532d89d9f26',
    level: 3,
    name: "Dragon's Breath",
    school: 'Evocation',
  },
  {
    id: 'a74d8c21-607d-4a3f-8cfd-2bbb48ec196a',
    level: 9,
    name: 'Wrath of the Titans',
    school: 'Evocation',
  },
];
