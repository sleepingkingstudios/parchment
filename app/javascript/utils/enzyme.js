import { valueOrDefault } from './object';
import { safeCapitalize } from './string';

export const extractInjectedComponent = (component) => {
  const injected = component.shallow().find('ComponentWithInjectedProps');

  return injected.exists() ? injected : component;
};

export const toHaveFormControl = expect => (received, expectedControl, withProps) => {
  const formControl = received.find(expectedControl);
  const expectedProps = valueOrDefault(withProps, {});

  expect(formControl).toExist();
  expect(formControl).toHaveProp(expectedProps);

  return { pass: true, message: () => '' };
};

export const toHaveFormGroup = expect => (received, expectedGroup, fn) => {
  const formGroup = received.find(expectedGroup);

  expect(formGroup).toExist();

  fn(extractInjectedComponent(formGroup).shallow());

  return { pass: true, message: () => '' };
};

export const toHaveFormInput = expect => (received, props) => {
  const {
    form,
    inputType,
    prop,
  } = props;

  expect(received).toHaveProp({ prop });

  const formInput = received.find(`${safeCapitalize(prop)}Input`);

  expect(formInput).toExist();
  expect(formInput).toHaveProp({ form });

  const shallowInput = formInput.shallow();

  expect(shallowInput).toExist();
  expect(shallowInput).toHaveDisplayName(valueOrDefault(inputType, 'FormInput'));

  return { pass: true, message: () => '' };
};
