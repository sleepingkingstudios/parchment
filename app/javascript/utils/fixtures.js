/* eslint-disable-next-line import/prefer-default-export */
export const trimMultilineLiteral = (text) => {
  if (!text) { return ''; }

  const paragraphs = text.split('\n\n');

  return paragraphs
    .map((paragraph) => {
      const lines = paragraph.split('\n');

      return lines.map(line => line.trim()).join(' ');
    })
    .join('\n\n');
};
