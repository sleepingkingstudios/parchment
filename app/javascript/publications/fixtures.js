import { load } from '../fixtures';

const isOfficial = (publication) => {
  const { playtest, publisherName } = publication;

  return !playtest && publisherName === 'Wizards of the Coast';
};

const rawData = load('publications');

/* eslint-disable-next-line import/prefer-default-export */
export const publicationsData = rawData
  .map(publication => Object.assign(
    {},
    publication,
    {
      official: isOfficial(publication),
    },
  ));
