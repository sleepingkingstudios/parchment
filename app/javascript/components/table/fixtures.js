import { humanizeList } from '../../utils/array';

export const formatCrewCapacity = (item) => {
  const count = item.crewCapacity;

  if (count === 1) { return '1 Kerbal'; }

  return `${count} Kerbals`;
};

export const formatExperiments = (item) => {
  humanizeList(item.experiments, { default: 'None' });
};

export const columns = [
  {
    label: 'Name',
    prop: 'name',
  },
  {
    label: 'Crew Capacity',
    prop: 'crewCapacity',
    value: formatCrewCapacity,
  },
  {
    label: 'Experiments',
    prop: 'experiments',
  },
];

export const rocketsData = [
  {
    id: '83f5e9e8-92df-4203-bc2e-4381370c2186',
    name: 'Apprentice',
    crewCapacity: 1,
    experiments: [
      'Crew Report',
      'Mystery Goo',
      'Temperature Scan',
      'Pressure Scan',
    ],
  },
  {
    id: '4bd7454c-3b15-4047-8d57-dfa831dac58e',
    name: 'Sorcerer',
    crewCapacity: 2,
    experiments: [
      'Crew Report',
      'Materials Study',
      'Mystery Goo',
      'Temperature Scan',
      'Pressure Scan',
    ],
  },
  {
    id: '2fead083-d9ea-4f04-949e-d8c00e800cbb',
    name: 'Warlock',
    crewCapacity: 3,
    experiments: [
      'Crew Report',
      'Materials Study',
      'Mystery Goo',
      'Temperature Scan',
      'Pressure Scan',
      'Gravity Scan',
    ],
  },
];
