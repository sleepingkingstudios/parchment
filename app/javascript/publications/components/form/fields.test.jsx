import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter as Router } from 'react-router-dom';

import {
  AbbreviationField,
  CancelButton,
  NameField,
  PlaytestField,
  PublicationDateField,
  PublisherNameField,
  SlugField,
  SubmitButton,
} from './fields';

import { publicationsData } from '../../fixtures';
import { INITIALIZED } from '../../../api/status';

describe('<PublicationForm /> fields', () => {
  const onChangeAction = jest.fn(
    ({ propName, value }) => ({ payload: { propName, value } }),
  );
  const onSubmitAction = jest.fn(() => ({ payload: { ok: true } }));
  const publication = publicationsData[0];
  const form = {
    data: { publication },
    path: ['publication'],
    onChangeAction,
    onSubmitAction,
  };

  describe('<AbbreviationField />', () => {
    const defaultProps = { form };

    it('should set the display name', () => {
      expect(AbbreviationField.displayName).toEqual('AbbreviationField');
    });

    it('should render a form field', () => {
      const rendered = shallow(<AbbreviationField {...defaultProps} />);

      expect(rendered).toHaveDisplayName('FormField');
    });

    it('should match the snapshot', () => {
      const rendered = mount(<AbbreviationField {...defaultProps} />);

      expect(rendered).toMatchSnapshot();
    });
  });

  describe('<CancelButton />', () => {
    const children = 'Do Nothing';
    const defaultProps = {
      children,
      form,
      isUpdate: false,
    };

    it('should set the display name', () => {
      expect(CancelButton.displayName).toEqual('CancelButton');
    });

    it('should render a form group', () => {
      const rendered = shallow(<CancelButton {...defaultProps} />);

      expect(rendered).toHaveDisplayName('FormGroup');
    });

    it('should match the snapshot', () => {
      const rendered = mount(<CancelButton {...defaultProps} />, { wrappingComponent: Router });

      expect(rendered).toMatchSnapshot();
    });
  });

  describe('<NameField />', () => {
    const defaultProps = { form };

    it('should set the display name', () => {
      expect(NameField.displayName).toEqual('NameField');
    });

    it('should render a form field', () => {
      const rendered = shallow(<NameField {...defaultProps} />);

      expect(rendered).toHaveDisplayName('FormField');
    });

    it('should match the snapshot', () => {
      const rendered = mount(<NameField {...defaultProps} />);

      expect(rendered).toMatchSnapshot();
    });
  });

  describe('<PlaytestField />', () => {
    const defaultProps = { form };

    it('should set the display name', () => {
      expect(PlaytestField.displayName).toEqual('PlaytestField');
    });

    it('should render a form field', () => {
      const rendered = shallow(<PlaytestField {...defaultProps} />);

      expect(rendered).toHaveDisplayName('FormField');
    });

    it('should match the snapshot', () => {
      const rendered = mount(<PlaytestField {...defaultProps} />);

      expect(rendered).toMatchSnapshot();
    });
  });

  describe('<PublicationDateField />', () => {
    const defaultProps = { form };

    it('should set the display name', () => {
      expect(PublicationDateField.displayName).toEqual('PublicationDateField');
    });

    it('should render a form field', () => {
      const rendered = shallow(<PublicationDateField {...defaultProps} />);

      expect(rendered).toHaveDisplayName('FormField');
    });

    it('should match the snapshot', () => {
      const rendered = mount(<PublicationDateField {...defaultProps} />);

      expect(rendered).toMatchSnapshot();
    });
  });

  describe('<PublisherNameField />', () => {
    const defaultProps = { form };

    it('should set the display name', () => {
      expect(PublisherNameField.displayName).toEqual('PublisherNameField');
    });

    it('should render a form field', () => {
      const rendered = shallow(<PublisherNameField {...defaultProps} />);

      expect(rendered).toHaveDisplayName('FormField');
    });

    it('should match the snapshot', () => {
      const rendered = mount(<PublisherNameField {...defaultProps} />);

      expect(rendered).toMatchSnapshot();
    });
  });

  describe('<SlugField />', () => {
    const defaultProps = { form };

    it('should set the display name', () => {
      expect(SlugField.displayName).toEqual('SlugField');
    });

    it('should render a form field', () => {
      const rendered = shallow(<SlugField {...defaultProps} />);

      expect(rendered).toHaveDisplayName('FormField');
    });

    it('should match the snapshot', () => {
      const rendered = mount(<SlugField {...defaultProps} />);

      expect(rendered).toMatchSnapshot();
    });
  });

  describe('<SubmitButton />', () => {
    const children = 'Do Something';
    const defaultProps = {
      children,
      form,
      isUpdate: false,
      status: INITIALIZED,
    };

    it('should set the display name', () => {
      expect(SubmitButton.displayName).toEqual('SubmitButton');
    });

    it('should render a form group', () => {
      const rendered = shallow(<SubmitButton {...defaultProps} />);

      expect(rendered).toHaveDisplayName('FormGroup');
    });

    it('should match the snapshot', () => {
      const rendered = mount(<SubmitButton {...defaultProps} />);

      expect(rendered).toMatchSnapshot();
    });
  });
});
