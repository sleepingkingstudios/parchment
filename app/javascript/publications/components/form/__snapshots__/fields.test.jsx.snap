// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PublicationForm /> fields <AbbreviationField /> should match the snapshot 1`] = `
<AbbreviationField
  form={
    Object {
      "data": Object {
        "publication": Object {
          "abbreviation": "ffh",
          "id": "7edf3a01-da19-475e-8e31-3f40c77acc62",
          "name": "The Flumph Fancier's Handbook",
          "official": true,
          "playtest": false,
          "publicationDate": "1977-05-25",
          "publisherName": "Wizards of the Coast",
          "slug": "flumph-fanciers-handbook",
        },
      },
      "onChangeAction": [MockFunction],
      "onSubmitAction": [MockFunction],
      "path": Array [
        "publication",
      ],
    }
  }
>
  <FormField
    colWidth={null}
    inputId={null}
    label={null}
    path={
      Array [
        "publication",
      ]
    }
    prop="abbreviation"
  >
    <FormGroup
      className="form-field"
      colWidth={null}
    >
      <div
        className="form-group col form-field"
      >
        <label
          htmlFor="publication-abbreviation-input"
        >
          Abbreviation
        </label>
        <AbbreviationInput
          form={
            Object {
              "data": Object {
                "publication": Object {
                  "abbreviation": "ffh",
                  "id": "7edf3a01-da19-475e-8e31-3f40c77acc62",
                  "name": "The Flumph Fancier's Handbook",
                  "official": true,
                  "playtest": false,
                  "publicationDate": "1977-05-25",
                  "publisherName": "Wizards of the Coast",
                  "slug": "flumph-fanciers-handbook",
                },
              },
              "onChangeAction": [MockFunction],
              "onSubmitAction": [MockFunction],
              "path": Array [
                "publication",
              ],
            }
          }
        >
          <FormInput
            className={null}
            id="publication-abbreviation-input"
            onChange={[Function]}
            placeholder=""
            type="text"
            validStatus={null}
            value="ffh"
          >
            <input
              className="form-control"
              id="publication-abbreviation-input"
              onChange={[Function]}
              placeholder=""
              type="text"
              value="ffh"
            />
          </FormInput>
        </AbbreviationInput>
      </div>
    </FormGroup>
  </FormField>
</AbbreviationField>
`;

exports[`<PublicationForm /> fields <CancelButton /> should match the snapshot 1`] = `
<CancelButton
  form={
    Object {
      "data": Object {
        "publication": Object {
          "abbreviation": "ffh",
          "id": "7edf3a01-da19-475e-8e31-3f40c77acc62",
          "name": "The Flumph Fancier's Handbook",
          "official": true,
          "playtest": false,
          "publicationDate": "1977-05-25",
          "publisherName": "Wizards of the Coast",
          "slug": "flumph-fanciers-handbook",
        },
      },
      "onChangeAction": [MockFunction],
      "onSubmitAction": [MockFunction],
      "path": Array [
        "publication",
      ],
    }
  }
  isUpdate={false}
>
  <FormGroup
    className=""
    colWidth={null}
  >
    <div
      className="form-group col"
    >
      <PublicationFormCancelButton
        form={
          Object {
            "data": Object {
              "publication": Object {
                "abbreviation": "ffh",
                "id": "7edf3a01-da19-475e-8e31-3f40c77acc62",
                "name": "The Flumph Fancier's Handbook",
                "official": true,
                "playtest": false,
                "publicationDate": "1977-05-25",
                "publisherName": "Wizards of the Coast",
                "slug": "flumph-fanciers-handbook",
              },
            },
            "onChangeAction": [MockFunction],
            "onSubmitAction": [MockFunction],
            "path": Array [
              "publication",
            ],
          }
        }
        isUpdate={false}
      >
        <LinkButton
          block={true}
          buttonSize={null}
          buttonStyle="secondary"
          className={null}
          link={false}
          outline={true}
          url="/publications"
        >
          <Link
            className="btn btn-outline-secondary btn-block"
            to="/publications"
          >
            <a
              className="btn btn-outline-secondary btn-block"
              href="/publications"
              onClick={[Function]}
            >
              Cancel
            </a>
          </Link>
        </LinkButton>
      </PublicationFormCancelButton>
    </div>
  </FormGroup>
</CancelButton>
`;

exports[`<PublicationForm /> fields <NameField /> should match the snapshot 1`] = `
<NameField
  form={
    Object {
      "data": Object {
        "publication": Object {
          "abbreviation": "ffh",
          "id": "7edf3a01-da19-475e-8e31-3f40c77acc62",
          "name": "The Flumph Fancier's Handbook",
          "official": true,
          "playtest": false,
          "publicationDate": "1977-05-25",
          "publisherName": "Wizards of the Coast",
          "slug": "flumph-fanciers-handbook",
        },
      },
      "onChangeAction": [MockFunction],
      "onSubmitAction": [MockFunction],
      "path": Array [
        "publication",
      ],
    }
  }
>
  <FormField
    colWidth={null}
    inputId={null}
    label={null}
    path={
      Array [
        "publication",
      ]
    }
    prop="name"
  >
    <FormGroup
      className="form-field"
      colWidth={null}
    >
      <div
        className="form-group col form-field"
      >
        <label
          htmlFor="publication-name-input"
        >
          Name
        </label>
        <NameInput
          form={
            Object {
              "data": Object {
                "publication": Object {
                  "abbreviation": "ffh",
                  "id": "7edf3a01-da19-475e-8e31-3f40c77acc62",
                  "name": "The Flumph Fancier's Handbook",
                  "official": true,
                  "playtest": false,
                  "publicationDate": "1977-05-25",
                  "publisherName": "Wizards of the Coast",
                  "slug": "flumph-fanciers-handbook",
                },
              },
              "onChangeAction": [MockFunction],
              "onSubmitAction": [MockFunction],
              "path": Array [
                "publication",
              ],
            }
          }
        >
          <FormInput
            className={null}
            id="publication-name-input"
            onChange={[Function]}
            placeholder=""
            type="text"
            validStatus={null}
            value="The Flumph Fancier's Handbook"
          >
            <input
              className="form-control"
              id="publication-name-input"
              onChange={[Function]}
              placeholder=""
              type="text"
              value="The Flumph Fancier's Handbook"
            />
          </FormInput>
        </NameInput>
      </div>
    </FormGroup>
  </FormField>
</NameField>
`;

exports[`<PublicationForm /> fields <PlaytestField /> should match the snapshot 1`] = `
<PlaytestField
  form={
    Object {
      "data": Object {
        "publication": Object {
          "abbreviation": "ffh",
          "id": "7edf3a01-da19-475e-8e31-3f40c77acc62",
          "name": "The Flumph Fancier's Handbook",
          "official": true,
          "playtest": false,
          "publicationDate": "1977-05-25",
          "publisherName": "Wizards of the Coast",
          "slug": "flumph-fanciers-handbook",
        },
      },
      "onChangeAction": [MockFunction],
      "onSubmitAction": [MockFunction],
      "path": Array [
        "publication",
      ],
    }
  }
>
  <FormField
    colWidth={null}
    inputId={null}
    label={null}
    path={
      Array [
        "publication",
      ]
    }
    prop="playtest"
  >
    <FormGroup
      className="form-field"
      colWidth={null}
    >
      <div
        className="form-group col form-field"
      >
        <label
          htmlFor="publication-playtest-input"
        >
          Playtest
        </label>
        <PlaytestInput
          form={
            Object {
              "data": Object {
                "publication": Object {
                  "abbreviation": "ffh",
                  "id": "7edf3a01-da19-475e-8e31-3f40c77acc62",
                  "name": "The Flumph Fancier's Handbook",
                  "official": true,
                  "playtest": false,
                  "publicationDate": "1977-05-25",
                  "publisherName": "Wizards of the Coast",
                  "slug": "flumph-fanciers-handbook",
                },
              },
              "onChangeAction": [MockFunction],
              "onSubmitAction": [MockFunction],
              "path": Array [
                "publication",
              ],
            }
          }
        >
          <FormCheckboxInput
            className={null}
            id="publication-playtest-input"
            label={null}
            onChange={[Function]}
            validStatus={null}
            value={false}
          >
            <div
              className="custom-control custom-checkbox"
            >
              <input
                checked={false}
                className="custom-control-input"
                id="publication-playtest-input"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor="publication-playtest-input"
              />
            </div>
          </FormCheckboxInput>
        </PlaytestInput>
      </div>
    </FormGroup>
  </FormField>
</PlaytestField>
`;

exports[`<PublicationForm /> fields <PublicationDateField /> should match the snapshot 1`] = `
<PublicationDateField
  form={
    Object {
      "data": Object {
        "publication": Object {
          "abbreviation": "ffh",
          "id": "7edf3a01-da19-475e-8e31-3f40c77acc62",
          "name": "The Flumph Fancier's Handbook",
          "official": true,
          "playtest": false,
          "publicationDate": "1977-05-25",
          "publisherName": "Wizards of the Coast",
          "slug": "flumph-fanciers-handbook",
        },
      },
      "onChangeAction": [MockFunction],
      "onSubmitAction": [MockFunction],
      "path": Array [
        "publication",
      ],
    }
  }
>
  <FormField
    colWidth={null}
    inputId={null}
    label={null}
    path={
      Array [
        "publication",
      ]
    }
    prop="publicationDate"
  >
    <FormGroup
      className="form-field"
      colWidth={null}
    >
      <div
        className="form-group col form-field"
      >
        <label
          htmlFor="publication-publication-date-input"
        >
          Publication Date
        </label>
        <PublicationDateInput
          form={
            Object {
              "data": Object {
                "publication": Object {
                  "abbreviation": "ffh",
                  "id": "7edf3a01-da19-475e-8e31-3f40c77acc62",
                  "name": "The Flumph Fancier's Handbook",
                  "official": true,
                  "playtest": false,
                  "publicationDate": "1977-05-25",
                  "publisherName": "Wizards of the Coast",
                  "slug": "flumph-fanciers-handbook",
                },
              },
              "onChangeAction": [MockFunction],
              "onSubmitAction": [MockFunction],
              "path": Array [
                "publication",
              ],
            }
          }
        >
          <FormInput
            className={null}
            id="publication-publication-date-input"
            onChange={[Function]}
            placeholder=""
            type="text"
            validStatus={null}
            value="1977-05-25"
          >
            <input
              className="form-control"
              id="publication-publication-date-input"
              onChange={[Function]}
              placeholder=""
              type="text"
              value="1977-05-25"
            />
          </FormInput>
        </PublicationDateInput>
      </div>
    </FormGroup>
  </FormField>
</PublicationDateField>
`;

exports[`<PublicationForm /> fields <PublisherNameField /> should match the snapshot 1`] = `
<PublisherNameField
  form={
    Object {
      "data": Object {
        "publication": Object {
          "abbreviation": "ffh",
          "id": "7edf3a01-da19-475e-8e31-3f40c77acc62",
          "name": "The Flumph Fancier's Handbook",
          "official": true,
          "playtest": false,
          "publicationDate": "1977-05-25",
          "publisherName": "Wizards of the Coast",
          "slug": "flumph-fanciers-handbook",
        },
      },
      "onChangeAction": [MockFunction],
      "onSubmitAction": [MockFunction],
      "path": Array [
        "publication",
      ],
    }
  }
>
  <FormField
    colWidth={null}
    inputId={null}
    label={null}
    path={
      Array [
        "publication",
      ]
    }
    prop="publisherName"
  >
    <FormGroup
      className="form-field"
      colWidth={null}
    >
      <div
        className="form-group col form-field"
      >
        <label
          htmlFor="publication-publisher-name-input"
        >
          Publisher Name
        </label>
        <PublisherNameInput
          form={
            Object {
              "data": Object {
                "publication": Object {
                  "abbreviation": "ffh",
                  "id": "7edf3a01-da19-475e-8e31-3f40c77acc62",
                  "name": "The Flumph Fancier's Handbook",
                  "official": true,
                  "playtest": false,
                  "publicationDate": "1977-05-25",
                  "publisherName": "Wizards of the Coast",
                  "slug": "flumph-fanciers-handbook",
                },
              },
              "onChangeAction": [MockFunction],
              "onSubmitAction": [MockFunction],
              "path": Array [
                "publication",
              ],
            }
          }
        >
          <FormInput
            className={null}
            id="publication-publisher-name-input"
            onChange={[Function]}
            placeholder=""
            type="text"
            validStatus={null}
            value="Wizards of the Coast"
          >
            <input
              className="form-control"
              id="publication-publisher-name-input"
              onChange={[Function]}
              placeholder=""
              type="text"
              value="Wizards of the Coast"
            />
          </FormInput>
        </PublisherNameInput>
      </div>
    </FormGroup>
  </FormField>
</PublisherNameField>
`;

exports[`<PublicationForm /> fields <SlugField /> should match the snapshot 1`] = `
<SlugField
  form={
    Object {
      "data": Object {
        "publication": Object {
          "abbreviation": "ffh",
          "id": "7edf3a01-da19-475e-8e31-3f40c77acc62",
          "name": "The Flumph Fancier's Handbook",
          "official": true,
          "playtest": false,
          "publicationDate": "1977-05-25",
          "publisherName": "Wizards of the Coast",
          "slug": "flumph-fanciers-handbook",
        },
      },
      "onChangeAction": [MockFunction],
      "onSubmitAction": [MockFunction],
      "path": Array [
        "publication",
      ],
    }
  }
>
  <FormField
    colWidth={null}
    inputId={null}
    label={null}
    path={
      Array [
        "publication",
      ]
    }
    prop="slug"
  >
    <FormGroup
      className="form-field"
      colWidth={null}
    >
      <div
        className="form-group col form-field"
      >
        <label
          htmlFor="publication-slug-input"
        >
          Slug
        </label>
        <SlugInput
          form={
            Object {
              "data": Object {
                "publication": Object {
                  "abbreviation": "ffh",
                  "id": "7edf3a01-da19-475e-8e31-3f40c77acc62",
                  "name": "The Flumph Fancier's Handbook",
                  "official": true,
                  "playtest": false,
                  "publicationDate": "1977-05-25",
                  "publisherName": "Wizards of the Coast",
                  "slug": "flumph-fanciers-handbook",
                },
              },
              "onChangeAction": [MockFunction],
              "onSubmitAction": [MockFunction],
              "path": Array [
                "publication",
              ],
            }
          }
        >
          <FormInput
            className={null}
            id="publication-slug-input"
            onChange={[Function]}
            placeholder=""
            type="text"
            validStatus={null}
            value="flumph-fanciers-handbook"
          >
            <input
              className="form-control"
              id="publication-slug-input"
              onChange={[Function]}
              placeholder=""
              type="text"
              value="flumph-fanciers-handbook"
            />
          </FormInput>
        </SlugInput>
      </div>
    </FormGroup>
  </FormField>
</SlugField>
`;

exports[`<PublicationForm /> fields <SubmitButton /> should match the snapshot 1`] = `
<SubmitButton
  disabled={false}
  form={
    Object {
      "data": Object {
        "publication": Object {
          "abbreviation": "ffh",
          "id": "7edf3a01-da19-475e-8e31-3f40c77acc62",
          "name": "The Flumph Fancier's Handbook",
          "official": true,
          "playtest": false,
          "publicationDate": "1977-05-25",
          "publisherName": "Wizards of the Coast",
          "slug": "flumph-fanciers-handbook",
        },
      },
      "onChangeAction": [MockFunction],
      "onSubmitAction": [MockFunction],
      "path": Array [
        "publication",
      ],
    }
  }
  isUpdate={false}
  status="initialized"
>
  <FormGroup
    className=""
    colWidth={null}
  >
    <div
      className="form-group col"
    >
      <PublicationFormSubmitButton
        disabled={false}
        form={
          Object {
            "data": Object {
              "publication": Object {
                "abbreviation": "ffh",
                "id": "7edf3a01-da19-475e-8e31-3f40c77acc62",
                "name": "The Flumph Fancier's Handbook",
                "official": true,
                "playtest": false,
                "publicationDate": "1977-05-25",
                "publisherName": "Wizards of the Coast",
                "slug": "flumph-fanciers-handbook",
              },
            },
            "onChangeAction": [MockFunction],
            "onSubmitAction": [MockFunction],
            "path": Array [
              "publication",
            ],
          }
        }
        isUpdate={false}
        status="initialized"
      >
        <FormSubmitWrapper
          block={true}
          disabled={false}
          form={
            Object {
              "data": Object {
                "publication": Object {
                  "abbreviation": "ffh",
                  "id": "7edf3a01-da19-475e-8e31-3f40c77acc62",
                  "name": "The Flumph Fancier's Handbook",
                  "official": true,
                  "playtest": false,
                  "publicationDate": "1977-05-25",
                  "publisherName": "Wizards of the Coast",
                  "slug": "flumph-fanciers-handbook",
                },
              },
              "onChangeAction": [MockFunction],
              "onSubmitAction": [MockFunction],
              "path": Array [
                "publication",
              ],
            }
          }
          outline={true}
        >
          <Button
            block={true}
            buttonSize="md"
            buttonStyle="primary"
            className={null}
            disabled={false}
            link={false}
            onClick={[Function]}
            outline={true}
          >
            <button
              className="btn btn-outline-primary btn-block"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Create Publication
            </button>
          </Button>
        </FormSubmitWrapper>
      </PublicationFormSubmitButton>
    </div>
  </FormGroup>
</SubmitButton>
`;
